.github/scripts/update_directory_tree.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+# This script updates the directory tree and saves it as a code block in a new Markdown file which is located inside the Contributor Notes/Structure folder.
+
+cd ../..
+
+OUTPUT_FILE="00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.01 Structure/Hub Tree Structure.md"
+
+tree -d -N -I venv --noreport > "$OUTPUT_FILE"
+sed -i "1i%% Hub Tree: Don’t edit below %%\n\`\`\`" "$OUTPUT_FILE" & \
```

Comment:
**On reflection, I'm totally comfortable with the comments below being ignored, and the script not changed, but I left them in for info...**

I'm not sure whether the following matters, since the script will now entirely be run via GitHub Action on Linux, but I did want to test it out for myself....

(And we already have some scripts that cannot be run on Windows anyway, so excluding Mac too isn't the end of the world!!)

Running this script on Mac, I get this output:

```
./update_directory_tree.sh 
sed: 1: "00 - Contribute to the  ...": invalid command code -                                                                                                   
```

Mac's sed is funny in how it handles `-I`

I wonder whether the code could be made to not need sed, perhaps by doing

```bash
echo 'first line' > "$OUTPUT_FILE"
tree ... >> "$OUTPUT_FILE"
echo "last line" >> "$OUTPUT_FILE"
```

Reply:
Trivia, but the backticks don't need to be escaped if you use single quotes rather than double quotes... But I don't know what that does the `\n` bits...

Reply:
Thanks for your comments. :) I wasn't home yesterday, so I only just now got around to reading everything. I could make it Mac compatible for testing purposes, that is true. But I didn't/don't have any way to test it, since I only use Ubuntu or Fedora.

The `\n` is there to get a new line, that's why it's also necessary to add the `-e` argument to `echo`. But you're right, I could've just added the first line with an echo. I sometimes have a tendency to make matters more complicated than they are. :sweat_smile: 

Should we change it to something like this? Or leave it?

```bash
echo -e '%% Hub Tree: Don’t edit below %%\n```' > "$OUTPUT_FILE"
tree -d -N -I venv --noreport >> "$OUTPUT_FILE"
echo -e '```\n%% Hub Tree: Don’t edit above %%' >> "$OUTPUT_FILE"
```

Reply:
Totally up to you... I don't mind either way.